CMakeLists.txt

```diff
@@ -48,12 +48,12 @@ if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ggdb -DDALI_DEBUG=1")
   set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -g -G -DDALI_DEBUG=1")
 else()
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DDALI_DEBUG=0")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -O2 -DDALI_DEBUG=0")
   set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -DDALI_DEBUG=0")
 endif()
 
 # CXX flags
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Werror -Wno-unused-variable -Wno-unused-function -fno-strict-aliasing -O2 -fPIC -fvisibility=hidden")
```

Comment:
I can discuss regarding -O2 but -Werror should stay where it is.

Reply:
In my opinion, we could treat debug configuration as development configuration also (I don't know, if its common thing). That way we could develop without need to pay attention to minor issues, but perform proper tests on release configuration, where everything will be in tact. I would imagine, that's the philosophy behind calling warning a warning, not an error. What's your opinion on that?

Reply:
My experience shows that unless your build breaks people tends to slips some warning from time to time and someone else needs to fix that latter and they tend to stack up.

Reply:
But in this case build _will_ break, since CI builds (and performs tests) on Release configuration

Reply:
@ptrendx - how much we don't want 02 in debug?

Reply:
Not sure actually, normally I would say we probably don't want it in debug build since it can make some values inaccessible in the debugger, but it never gave me any trouble in DALI.

Reply:
@ptrendx  And how would you feel about `-Werror` flag for Debug configuration?

Reply:
I'm not sure you gave a compelling reason to omit it for debug builds.
Personally, I build release all the time and debug only if something segfaults and I need better backtrace from debugger, so it does not affect me. However I can see somebody building mainly debug builds and then unnecessarily spending time debugging issue coming from uninitialized variable or something that `-Werror` would prevent.

Reply:
It is not like you cannot disable -Werror locally in your CMake for the time of development and later revert it.

Reply:
I agree with @klecki  and @ptrendx, leave it as it was, if someone like to hunt problems in debug build which can be easily detected by ```-Werror``` he can change this setting locally in his workspace.